scripts/tools/CleanUpOrphanFiles.php

```diff
@@ -0,0 +1,239 @@
+            ],
+            'verbose' => [
+                'prefix' => 'v',
+                'flag' => true,
+                'longPrefix' => 'Verbose',
+                'description' => 'Force script to be more details',
+            ],
+
+        ];
+    }
+
+    protected function provideDescription()
+    {
+        return 'Tool to remove orphan files attached to removed items. By default in dry-run';
+    }
+
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints a help statement'
+        ];
+    }
+
+    /**
+     * @return Report
+     * @throws \oat\oatbox\service\exception\InvalidServiceManagerException
+     * @throws \common_exception_Error
+     */
+    protected function run()
+    {
+        $this->init();
+
+        $this->report = Report::createInfo('Following files');
+
+        /** @var ResourceFileSerializer $serializer */
+        $serializer = $this->getServiceManager()->get(ResourceFileSerializer::SERVICE_ID);
+        $total = 0;
+        $resourceCollection = new ResourceCollection(GenerisRdf::CLASS_GENERIS_FILE);
+        foreach ($resourceCollection as $resource) {
+            $total++;
+            try {
+                $subject = $this->getResource($resource['subject']);
+                $file = $serializer->unserialize($subject);
+                $isRedundant = $this->isRedundant($file);
+
+                if ($isRedundant) {
+                    $this->manageRedundant($subject, $file);
+                } else {
+                    $this->manageOrphan($subject, $file);
+                }
+
+            } catch (\Exception $exception) {
+                $this->errorsCount++;
+                $this->report->add(Report::createFailure($exception->getMessage()));
+            }
+        }
+
+        $this->cleanUp();
+
+        $this->report->add(new Report(Report::TYPE_SUCCESS, sprintf('%s Total Files Found in RDS, where: ', $total)));
+
+        $this->prepareReport();
+
+        return $this->report;
+    }
+
+    private function init()
+    {
+        if ($this->getOption('wet-run')) {
+            $this->wetRun = true;
+        }
+        $this->verbose = $this->getOption('verbose');
+
+    }
+
+    protected function showTime()
+    {
+        return true;
+    }
+
+    /**
+     * @param core_kernel_classes_Resource $resource
+     * @return bool
+     */
+    private function isOrphan(core_kernel_classes_Resource $resource)
+    {
+        $sql = 'SELECT subject FROM statements s WHERE s.object=?';
+        $stmt = $this->getPersistence()->query($sql, [$resource->getUri()]);
+        $res = $stmt->fetchAll();
+
+        return 0 === count($res);
+    }
+
+    private function getPersistence()
+    {
+        return $this->getServiceLocator()
+            ->get(\common_persistence_Manager::SERVICE_ID)
+            ->getPersistenceById('default');
+    }
+
+    private function getRedundantFiles()
```

Comment:
Pls pay attention this method isn't used, ignore my notice if it's ok (I'm not aware of the whole task)